packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -265,6 +275,9 @@ function hydrateComment(node: Node, vnode: VComment, renderer: RendererAPI): Nod
 }
 
 function hydrateStaticElement(elm: Node, vnode: VStatic, renderer: RendererAPI): Node | null {
+    if (shouldIgnoreAllMismatches(elm, renderer)) {
+        return elm;
+    }
```

Comment:
This is wrong because we need to just skip mismatch validation, not all of hydration.